src/services/ticket.service.ts

```diff
@@ -0,0 +1,40 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {environment} from '../environments/environment';
+import {ServerMessage} from '../model/server-message';
+import {Router} from '@angular/router';
+
+const COULD_NOT_CONNECT = 'Could not connect to server.';
+const TOKEN_NAME = 'ACCESS_TOKEN';
+
+
+@Injectable()
+export class TicketService {
+
+  constructor(private http: HttpClient, private router: Router) {}
+
+  createTicket(subject: string, message: string,
+                next?: (msg: ServerMessage) => void): void {
+
+    this.http.post<ServerMessage>(environment.serverUrl + '/create-ticket',
+      {subject: subject, message: message})
+      .subscribe(
+        res => {
+          next(res);
+        }, err => {
+          next(new ServerMessage(false, COULD_NOT_CONNECT));
+        }
+      );
+  }
+
+  authenticate(next: (isAuthenticated: boolean) => void): void {
```

Comment:
You do not need to duplicate this method. If you need to call authenticate for the contact page, just use the `AccountService`'s `authenticate()` method.